src/imperator/tables/triggers.rs

```diff
@@ -0,0 +1,526 @@
+use crate::everything::Everything;
+use crate::item::Item;
+use crate::scopes::*;
+use crate::token::Token;
+use crate::trigger::{RawTrigger, Trigger};
+
+use RawTrigger::*;
+
+pub fn scope_trigger(name: &Token, data: &Everything) -> Option<(Scopes, Trigger)> {
+    let name_lc = name.as_str().to_lowercase();
+
+    // TODO: binary search might be faster
+    if let Some((from, trigger)) = TRIGGER_MAP.get(&name_lc) {
+        return Some((*from, trigger.clone()));
+    }
+    if let Some(conviction) = name_lc.strip_suffix("_conviction") {
+        data.verify_exists_implied(Item::PartyType, conviction, name);
+        return Some((Scopes::Character, Trigger::CompareValue)));
+    }
+    if let Some(support) = name_lc.strip_suffix("_support") {
+        data.verify_exists_implied(Item::PartyType, support, name);
+        return Some((Scopes::Country, Trigger::CompareValue)));
+    }
+    if let Some(happiness) = name_lc.strip_suffix("_happiness") {
+        data.verify_exists_implied(Item::PopType, happiness, name);
+        return Some((Scopes::Province, Trigger::CompareValue)));
+    }
+    if let Some(part) = name.as_str().strip_prefix("num_of_") {
+        if warn
+            && !data.item_exists(Item::Building, part)
+            && !data.item_exists(Item::PopType, s)
+        {
+            let msg = format!("could not find any {part}");
+            let info = "Possible valid options would be: num_of_$POPTYPE$ or num_of_$BUILDING$";
+            warn_info(name, ErrorKey::MissingItem, &msg, info);
+        }
+        return Some((Scopes::Country, Trigger::CompareValue)));
+    }
+    // This one is weird...the trigger is just Item::TechnologyTable with no suffix or prefix.
+    if data.verify_exists_implied(Item::TechnologyTable, "", name) {
+        return Some((Scopes::Country, Trigger::CompareValue)));
+    }
+    std::option::Option::None
+}
+
+static TRIGGER_MAP: Lazy<FnvHashMap<String, (Scopes, Trigger)>> = Lazy::new(|| {
+    let mut hash = FnvHashMap::default();
+    for (from, s, trigger) in TRIGGER {
+        hash.insert(
+            (*s).to_string(),
+            (Scopes::from_bits_truncate(*from), Trigger::from_raw(trigger)),
+        );
+    }
+    hash
+});
+
+/// LAST UPDATED VIC3 VERSION 1.3.6
+/// See `triggers.log` from the game data dumps
+/// A key ends with '(' if it is the version that takes a parenthesized argument in script.
+const TRIGGER: &[(u64, &str, RawTrigger)] = &[
```

Comment:
This needs changes too: `u64` to `Scopes`, and `RawTrigger` to just `Trigger`, and every Scope value in the table should start with `Scopes::`

All to get rid of `RawTrigger` :)